### PR TITLE
Blog app: add authorization rules

### PR DESCRIPTION
**Pull request by denisdiaconu**
In this project, we added authorization to our app using CanCanCan.
We did the following:

-  Install CanCanCan in your project.
-  Add a role column to the users table. Remember to use a migration for this.
-  A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.

-    For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that anlu authorized users can see it.

- A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.

 -   For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that anlu authorized users can see it.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -2,11 +2,25 @@ class PostsController < ApplicationController
   def index
```
Comment (GzimAsani) [inline app/controllers/posts_controller.rb:3]: - [ ] You should use **CanCanCan** for Post authorization in `Post Controller`
example:

```ruby
 class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```

File: Gemfile [hunk 2]
```diff
@@ -54,6 +54,7 @@ group :test do
   gem 'webdrivers'
 end
 
+gem 'cancancan'
```
Comment (GzimAsani) [inline Gemfile]: - [ ] Please add a more detailed description in your Pull Request

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# Hi @denisdiaconu  and @hiromataba                             
Good job so far! Your code looks fine, good job :+1:  :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight:
- Good job with the gitflow :+1:
- Good PR description :clap: 

## Required Changes ♻️ 
- [ ] _Check inline comments for more information :arrow_double_down:_ 


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by arslanbisharat — APPROVED** [review]
# Approved 👏 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

To Highlight: 
- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of code reviews per project which you can see from your dashboard. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

**Pull request merged**